### DISCUSSION TITLE
What is the maximum timeout value in Azure APIM policy?

### DISCUSSION THREAD

**Discussion by Panchadurga** [Azure/apim-lab#121] [Q&A]
I have set the APIM timeout policy in the backend section to 60 seconds. I am facing an issue where a client who is calling my APIM endpoint receives a 504 gateway error due to timeout at forward-request (after 60 seconds)

**Note** : Default timeout value is 300 secs as per Microsoft document and please let us know the **maximum timeout value** in Azure APIM policies.

**Comment by Panchadurga** [comment]
Is it possible to increase the timeout value at the APIM policy level to prevent the gateway timeout error?

**Comment by simonkurtz-MSFT** [comment]
Hi @Panchadurga, I am not positive I am fully understanding your question. If you are setting the `timeout` property in `forward-request` to be 60 seconds, and you then receive a timeout after 60 seconds, is it not working as expected? 

While a maximum of 300 seconds exists, the underlying networking infrastructure will almost certainly time out requests just shy of 240 seconds. This is an Azure-wide configuration and not something you can increase. 

Can you tell me what your use case is for a long-running request, please? I might be able to make some suggestions if I have more knowledge. Also, is Azure APIM the first hop in the call chain or do you have an Azure App Gateway, Azure Front Door, etc. between Azure APIM and the caller?

**Reply by Panchadurga** [reply]
Hi @simonkurtz-MSFT ,

Yes, it is for a long-running request with a large payload, and you are correct that Azure APIM is the first hop in the call chain. However, it is taking longer than 5 minutes to receive a response from the backend service running on the Azure Spring cloud.

Please provide suggestions on how to avoid timeout problems at the APIM level or the Azure Spring cloud level.

Thanks

**Reply by simonkurtz-MSFT** [reply]
Hi @Panchadurga,

Thank you for those details. Since your backend is taking more time than the Azure infrastructure's network timeouts, you would need to consider alternatives. Perhaps changing the API to initiate a request, returning quickly, then using polling for status (similar to what Azure Durable Functions do) would suit your needs. 

Also be careful with large payloads in Azure APIM, especially if you act upon those payloads as that becomes quite costly with respect to memory.

**Reply by Panchadurga** [reply]
Thank you

**Comment by jakeatmsft** [comment]
@simonkurtz-MSFT - Does the 240sec limit still apply to APIM?  We are using APIM with LLM backend and some models can take over 5min to respond due to model processing time, can the timeout be increased in this case?  Would streaming responses (SSE) be a workaround?
